solana/src/main/java/com/solana/vendor/bip39/Mnemonic.kt

```diff
@@ -0,0 +1,271 @@
+package com.solana.vendor.bip39
+
+import com.solana.vendor.bip39.Mnemonics.MnemonicCode
+import java.io.Closeable
+import java.security.MessageDigest
+import java.security.SecureRandom
+import java.util.*
+
+import kotlin.experimental.or
+
+class Mnemonic(
+    val phrase: List<String>,
+    val passphrase: String = ""
+) {
+    constructor(wordCount: WordCount, passphrase: String = ""): this(
+        MnemonicCode(wordCount).words.map { it.joinToString(separator = "") { word -> "$word" } },
+        passphrase
+    )
+
+    @Throws(RuntimeException::class)
```

Comment:
Wonder if it would be better to to put all the errors under 1 super class (sealed class) and specify that class here, rather than RuntimeException. Thoughts?